src/Gui/Inventor/Draggers/SoLinearDragger.cpp

```diff
@@ -216,10 +207,19 @@ SoSeparator* SoLinearDragger::buildLabelGeometry()
 
 SoBaseColor* SoLinearDragger::buildActiveColor()
 {
-    auto colorActive = new SoBaseColor();
-    colorActive->rgb.setValue(1.0, 1.0, 0.0);
+    auto color = new SoBaseColor;
+    color->rgb.connectFrom(&activeColor);
 
-    return colorActive;
+    return color;
+}
+
+void SoLinearDragger::setupGeometryCalculator()
+{
+    calculator = new SoCalculator;
+    calculator->expression =
+        "oA = vec3f(0, a * 0.5, 0); "
+        "oB = vec3f(0, a + b * 0.5, 0); "
+        "oC = vec3f(0, a + b * 1.5, 0); ";
```

Comment:
That seems odd to me. What does it calculate, what does "a", "b" and "c" mean?

Reply:
That is basically the open inventor way of writing expressions that will be used to calculate the output whenever one of the input field changes. If I avoid this then I will need to write a lot of sensor related code.

Reply:
https://developer9.openinventor.com/UserGuides/9.9/Inventor_Mentor/Engines/Arithmetic_Engines.html

Reply:
I have added some comments. Can you please take a look if it is sufficient or not?